Client/Frontend/Browser/Tab.swift

```diff
@@ -679,3 +682,66 @@ class TabWebViewMenuHelper: UIView {
         }
     }
 }
+
+// MARK: - Brave Search
+
+extension Tab {
+    /// Call the api on the Brave Search website and passes the fallback results to it.
+    /// Important: This method is also called when there is no fallback results
+    /// or when the fallback call should not happen at all.
+    /// The website expects the iOS device to always call this method(blocks on it).
+    func injectResults() {
+        let dispatchGroup = DispatchGroup()
+        var shouldInjectResults = true
+        
+        dispatchGroup.enter()
+        DispatchQueue.main.async {
+            // If the backup search results happen before the Brave Search loads
+            // The method we pass data to is undefined.
+            // For such case we do not call that method or remove the search backup manager.
+            // swiftlint:disable:next safe_javascript
+            self.webView?.evaluateJavaScript("window.onFetchedBackupResults === undefined") {
+                result, error in
+                defer { dispatchGroup.leave() }
+                
+                if let error = error {
+                    log.error("onFetchedBackupResults existence check error: \(error)")
+                }
+                
+                guard let resultBool = result as? Bool else {
+                    log.error("onFetchedBackupResults existence check, failed to unwrap bool result value")
+                    return
+                }
+                
+                if resultBool {
+                    shouldInjectResults = false
+                }
+            }
+        }
+        
+        dispatchGroup.notify(queue: .main) { [weak self] in
+            guard let self = self else { return }
+            if !shouldInjectResults {
+                log.info("Search Backup results are ready but the page has not been loaded yet")
+                return
+            }
+                
+            var queryResult = "null"
+            
+            if let url = self.webView?.url,
+               BraveSearchManager.isValidURL(url),
+               let result = self.braveSearchManager?.fallbackQueryResult {
+                queryResult = result
+            }
+            
+            self.webView?.evaluateSafeJavaScript(
+                functionName: "window.onFetchedBackupResults",
+                args: [queryResult],
+                sandboxed: false,
+                escapeArgs: false)
```

Comment:
i think this should be escaping args and sandboxing; otherwise Google can cause XSS on Brave Search with their results. cc @jumde

Reply:
I do not think this can be sandboxed if we want to communicate from native to website-side.
I tried escapeArgs param but it failed to pass data, that's why i had to add my custom method for escaping 'javascript'